modules/pam_namespace/pam_namespace.c

```diff
@@ -39,6 +39,71 @@
 #include "pam_namespace.h"
 #include "argv_parse.h"
 
+/* --- evaluting all files in VENDORDIR/security/namespace.d and /etc/security/namespace.d --- */
+static int
+compare_filename(const void *a, const void *b)
+{
+	char *a_dup = strdup(*(const char * const *)a);
+	char *b_dup = strdup(*(const char * const *)b);
+	int ret = strcmp(basename(a_dup),
+			 basename(b_dup));
+	free(a_dup);
+	free(b_dup);
+	return ret;
+}
+
+/* Evaluating a list of files which have to be parsed in the right order:
+ *
+ * - If etc/security/namespace.d/@filename@.conf exists, then
+ *   %vendordir%/security/namespace.d/@filename@.conf should not be used.
+ * - All files in both namespace.d directories are sorted by their @filename@.conf in
+ *   lexicographic order regardless of which of the directories they reside in. */
+static char **read_access_dir(void)
+{
+	glob_t globbuf;
+	int i=0;
+	int glob_rv = glob(NAMESPACE_D_GLOB, GLOB_ERR, NULL, &globbuf);
+#ifdef VENDOR_NAMESPACE_D_GLOB
+	glob_t globbuf_vendor;
+	int glob_rv_vendor = glob(VENDOR_NAMESPACE_D_GLOB, GLOB_ERR, NULL, &globbuf_vendor);
+	char **file_list = malloc((globbuf_vendor.gl_pathc + globbuf.gl_pathc + 1) * sizeof(char*));
+#else
+	char **file_list = malloc((globbuf.gl_pathc + 1) * sizeof(char*));
+#endif
+
+	if (!glob_rv) {
+	    for (i = 0; globbuf.gl_pathv[i] != NULL; i++) {
```

Comment:
I think it is NULL terminated. Otherwise all my tests would have been failed. But the docu says nothing about it. So, you are right to use gl_pathc instead.